include/zephyr/bluetooth/classic/avrcp.h

```diff
@@ -0,0 +1,88 @@
+/** @file
+ *  @brief Audio Video Remote Control Profile header.
+ */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright (C) 2024 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AVRCP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AVRCP_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief AVRCP structure */
+struct bt_avrcp;
+
+struct bt_avrcp_cb {
+	/** @brief An AVRCP connection has been established.
+	 *
+	 *  This callback notifies the application of an avrcp connection,
+	 *  i.e., an AVCTP L2CAP connection.
+	 *
+	 *  @param avrcp AVRCP connection object.
+	 */
+	void (*connected)(struct bt_avrcp *avrcp);
+	/** @brief An AVRCP connection has been disconnected.
+	 *
+	 *  This callback notifies the application that an avrcp connection
+	 *  has been disconnected.
+	 *
+	 *  @param avrcp AVRCP connection object.
+	 */
+	void (*disconnected)(struct bt_avrcp *avrcp);
+};
+
+/** @brief Connect AVRCP.
+ *
+ *  This function is to be called after the conn parameter is obtained by
+ *  performing a GAP procedure. The API is to be used to establish AVRCP
+ *  connection between devices.
+ *
+ *  @param conn Pointer to bt_conn structure.
+ *
+ *  @return pointer to struct bt_avrcp in case of success or NULL in case
+ *  of error.
+ */
+struct bt_avrcp *bt_avrcp_connect(struct bt_conn *conn);
+
+/** @brief Disconnect AVRCP.
+ *
+ *  This function close AVCTP L2CAP connection.
+ *
+ *  @param avrcp The AVRCP instance.
+ *
+ *  @return 0 in case of success or error code in case of error.
+ */
+int bt_avrcp_disconnect(struct bt_avrcp *avrcp);
+
+/** @brief Register callback.
+ *
+ *  Register AVRCP callbacks to monitor the state and interact with the remote device.
+ *
+ *  @param cb The callback function.
+ *
+ *  @return 0 in case of success or error code in case of error.
+ */
+int bt_avrcp_register_cb(struct bt_avrcp_cb *cb);
```

Comment:
Since this is a singleton (e.g. you don't have a linked list pointer in the struct), I think `const struct bt_avrcp_cb *cb` is probably more correct. That said, have you considered whether you want to keep this a singleton, i.e. that there aren't any valid use cases where an app might be interested in registering multiple instances of the callbacks?

Reply:
Hmm, technically the Bluetooth Spec allows multiple instances, however, none of the applications did so.
As for the target side, one AVRCP instance can control multiple players, thus multiple instances is meaningless.
As for the controller side, we don't really need two distinct control panels on one product.
I saw both A2DP and HFP assume singleton, I choose to follow. 'const' is not used as well at these profiles.

Reply:
> I saw both A2DP and HFP assume singleton, I choose to follow. 'const' is not used as well at these profiles.

We should then add `const` to those places too. Please don't assume that existing code must be "correct" just because it's already in the tree, especially in the case of Bluetooth Classic.

Reply:
> > I saw both A2DP and HFP assume singleton, I choose to follow. 'const' is not used as well at these profiles.
> 
> We should then add `const` to those places too. Please don't assume that existing code must be "correct" just because it's already in the tree, especially in the case of Bluetooth Classic.

Done.